vehicle/zero/provider.go

```diff
@@ -0,0 +1,108 @@
+package zero
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/provider"
+)
+
+// Provider implements the vehicle api
+type Provider struct {
+	api    *API
+	status provider.Cacheable[ZeroState]
+}
+
+// NewProvider creates a vehicle api provider
+func NewProvider(api *API, cache time.Duration) *Provider {
+	impl := &Provider{
+		api: api,
+		status: provider.ResettableCached(func() (ZeroState, error) {
+			return api.Status()
+		}, cache),
+	}
+
+	return impl
+}
+
+var _ api.Battery = (*Provider)(nil)
+
+// Soc implements the api.Vehicle interface
+func (v *Provider) Soc() (float64, error) {
+	res, err := v.status.Get()
+	if err != nil {
+		return 0, err
+	}
+
+	val := res.Soc
+
+	return float64(val), nil
+}
+
+var _ api.ChargeState = (*Provider)(nil)
+
+// Status implements the api.ChargeState interface
+func (v *Provider) Status() (api.ChargeStatus, error) {
+	res, err := v.status.Get()
+	if err != nil {
+		return api.StatusNone, err
+	}
+
+	status := api.StatusA // disconnected
+
+	if res.Pluggedin != 0 {
```

Comment:
My first version was more efficient in case nothing is plugged in, which is probably the case most of the time.
This is why I created the test as it was.